### ISSUE TITLE
Firebase-tools is not uploading newly created functions to Firebase

### ISSUE THREAD

**Issue by Brandon9598** [firebase/firebase-tools#1210]
### Environment info

**firebase-tools: 6.7.0**
**Platform: macOS**
**Angular CLI: 7.3.8**
### Steps to reproduce

Create a new function in the functions/src/index.ts file and run `firebase deploy --only functions`

Below is a code snippet of the index.ts file I have inside my Angular project:

```typescript
import * as functions from 'firebase-functions';
import { firestore } from 'firebase-admin';
import * as moment from 'moment';
import { request } from 'https';

// Local Functions Imports
import * as Permissions from './permissions';
import * as Groups from './groups';
import * as Shifts from './shifts';
import * as Roles from './roles';
import * as Session from './sessions';
import * as Users from './users';
import * as Slack from './slack';
import * as ScheduleChanges from './schedule-changes';
import * as Email from './email';

// The Firebase Admin SDK to access the Firebase Realtime Database.
const admin = require('firebase-admin');
admin.initializeApp(functions.config().firebase);

/* -------- USER MANAGEMENT -------------- */
export const createUser = Users.createUser;
export const addUserToDefaultJoinGroups = Users.addUserToDefaultJoinGroups;
export const deleteUser = Users.deleteUser;

/* -------- SESSION MANAGEMENT -------------- */
export const addSessionID = Session.addSessionID;

/* -------- ROLE MANAGEMENT -------------- */
export const addRoleID = Roles.addRoleID;
export const removeAllShiftsAssociatedWithRole = Roles.removeAllShiftsAssociatedWithRole;

/* -------- SHIFT MANAGEMENT -------------- */
export const addShiftID = Shifts.addShiftID;
export const hourly_job = Shifts.hourly_job;
export const changeShiftStatsTest = Shifts.changeShiftStatsTest;

/* -------- GROUPS MANAGEMENT -------------- */
export const addGroupID = Groups.addGroupID;
export const addGroupIDTest = Groups.addGroupIDNewTestFunction;

/* -------- PERMISSIONS MANAGEMENT -------------- */
export const addPermissionID = Permissions.addPermissionID;

/* -------- Emailing -------------- */
export const sendWelcomeEmailToNewUser = Email.sendWelcomeEmailToNewUser;
export const sendWelcomeEmailToNewUser2 = Email.sendWelcomeEmailToNewUser2;

/* -------- SLACK MESSAGING MANAGEMENT -------------- */
export const sendWelcomingMessage = Slack.sendWelcomingMessage;

/* -------- SCHEDULE CHANGES MANAGEMENT -------------- */
export const addScheduleChangeID = ScheduleChanges.addScheduleChangeID;

```

### Expected behavior
As seen in the code above, there are 16 functions that should be deployed to Firebase when I run `firebase deploy --only functions`. 

### Actual behavior

However, only the 12 existing functions that are already in Firebase get deployed. The following code snippet is from functions/src/groups/index.ts. In this file, the addGroupID is an existing function which continually gets deployed to Firebase. On the other hand, addGroupIDNewTestFunction is a new function that isn't getting deployed to Firebase even though they are in the same file and being referenced the same way.

```typescript
import * as functions from 'firebase-functions';
import * as admin from 'firebase-admin'

export const addGroupID = functions.firestore
.document("organizations/{organizationID}/groups/{groupID}")
.onCreate(async (snap, context) => {
    console.log("Adding group id");
    await admin.firestore()
    .doc(`organizations/${context.params.organizationID}/groups/${context.params.groupID}`).update({
        'groupID': context.params.groupID
    })
})

export const addGroupIDNewTestFunction = functions.firestore
.document("organizations/{organizationID}/groups2/{groupID}")
.onCreate(async (snap, context) => {
    console.log("Adding group id");
    await admin.firestore()
    .doc(`organizations/${context.params.organizationID}/groups/${context.params.groupID}`).update({
        'groupID': context.params.groupID
    })
})
```
The following code snippet is the output that comes from running `firebase deploy --only functions`:

```bash
firebase deploy --only functions

=== Deploying to 'university-scheduling'...

i  deploying functions
Running command: npm --prefix "$RESOURCE_DIR" run lint

> functions@ lint /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions
> tslint --project tsconfig.json

no-unused-variable is deprecated. Since TypeScript 2.9. Please use the built-in compiler checks instead.

Could not find implementations for the following rules specified in the configuration:
    use-input-property-decorator
    use-output-property-decorator
    use-host-property-decorator
Try upgrading TSLint and/or ensuring that you have all necessary custom rules installed.
If TSLint was recently upgraded, you may have old rules configured which need to be cleaned up.


WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/email/index.ts:2:13 - 'admin' is declared but its value is never read.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/index.ts:2:1 - All imports on this line are unused.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/index.ts:3:13 - 'moment' is declared but its value is never read.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/index.ts:4:1 - All imports on this line are unused.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/slack/index.ts:2:13 - 'admin' is declared but its value is never read.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/slack/index.ts:6:7 - 'request' is declared but its value is never read.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/slack/index.ts:7:7 - 'options' is declared but its value is never read.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/users/index.ts:3:1 - All imports on this line are unused.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/users/index.ts:43:11 - 'groups' is declared but itsvalue is never read.
WARNING: /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions/src/users/index.ts:65:11 - 'uid' is declared but its value is never read.

Running command: npm --prefix "$RESOURCE_DIR" run build

> functions@ build /Users/brandon/Dropbox/Bearforce_Scheduling/bearforce-website/functions
> tsc

✔  functions: Finished running predeploy script.
i  functions: ensuring necessary APIs are enabled...
✔  functions: all necessary APIs are enabled
i  functions: preparing functions directory for uploading...
i  functions: packaged functions (106.24 KB) for uploading
✔  functions: functions folder uploaded successfully
⚠  appEngineLocation us-central1
i  functions: updating Node.js 6 function createUser(us-central1)...
i  functions: updating Node.js 6 function deleteUser(us-central1)...
i  functions: updating Node.js 6 function addSessionID(us-central1)...
i  functions: updating Node.js 6 function addRoleID(us-central1)...
i  functions: updating Node.js 6 function removeAllShiftsAssociatedWithRole(us-central1)...
i  functions: updating Node.js 6 function addShiftID(us-central1)...
i  functions: updating Node.js 6 function hourly_job(us-central1)...
i  functions: updating Node.js 6 function changeShiftStatsTest(us-central1)...
i  functions: updating Node.js 6 function addGroupID(us-central1)...
i  functions: updating Node.js 6 function addPermissionID(us-central1)...
i  functions: updating Node.js 6 function sendWelcomingMessage(us-central1)...
i  functions: updating Node.js 6 function addScheduleChangeID(us-central1)...
✔  scheduler: all necessary APIs are enabled
✔  functions[addPermissionID(us-central1)]: Successful update operation.
✔  functions[createUser(us-central1)]: Successful update operation.
✔  functions[addShiftID(us-central1)]: Successful update operation.
✔  functions[changeShiftStatsTest(us-central1)]: Successful update operation.
✔  functions[sendWelcomingMessage(us-central1)]: Successful update operation.
✔  functions[hourly_job(us-central1)]: Successful update operation.
✔  functions[addSessionID(us-central1)]: Successful update operation.
✔  functions[deleteUser(us-central1)]: Successful update operation.
✔  functions[addGroupID(us-central1)]: Successful update operation.
✔  functions[removeAllShiftsAssociatedWithRole(us-central1)]: Successful update operation.
✔  functions[addScheduleChangeID(us-central1)]: Successful update operation.
✔  functions[addRoleID(us-central1)]: Successful update operation.

✔  Deploy complete!

Please note that it can take up to 30 seconds for your updated functions to propagate.
```
As seen in the code above, only the 12 existing functions which are already in Firebase get updated and the other 4 are ignored.

### Attempted Solutions
1. I made sure that I am on the latest firebase-tools (6.7.0)
2. I have tried specifying the newly written functions I would like to deploy using firebase deploy --only functions:<NEW FUNCTION NAME HERE>
3. Remove all export functions from the functions/src/index.ts file and upload a blank index.ts. Even when using a blank index.ts file, the same functions still get deployed to Firebase.
4. I know there can be up to a 30-second delay when uploading functions to Firebase, so I have even waited for an entire night before trying to upload new functions.
5. I have started from complete scratch using firebase init to recreate the whole functions directory and it still keeps uploading the same functions.
6. I upgraded from the free Firebase plan to the Spark plan.

**Comment by rilian** [comment]
maybe cause nodejs 6 is deprecated and after 22th Apr it wont allow to deploy them https://cloud.google.com/functions/docs/concepts/nodejs-6-runtime

**Comment by bkendall** [comment]
Can you verify for me that in your compiled code (which is in the folder that is actually being deployed) you see all 16 of the functions you are expecting to be deployed?

I don't believe that deploys to Node 6 functions are blocked yet, but do know that migrating to Node 8 will be required soon. If you'd like to try [specifying the node 6 or 8 runtime](https://firebase.google.com/docs/functions/manage-functions#set_nodejs_version) in your `package.json`, that might be a good reminder. I don't think omitting that value should prevent the deploy, but maybe setting it to `6` might help?

**Comment by Brandon9598** [comment]
I wanted to give you guys an update on the solution to the problem I described above. In one of my index.ts files, I had the following require statement, `var mailgun = require('mailgun-js')`.  Whenever I replaced this the require statement with `import * as mailgun from 'mailgun-js'`, all the functions were deployed properly. 

The most confusing part about this whole issue was that I only received success messages even when things weren't working properly. I do thank you guys for the speedy support!!
